internal/repository/file.go

```diff
@@ -0,0 +1,265 @@
+package repository
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/storage"
+	"fyne.io/fyne/storage/repository"
+)
+
+// declare conformance with repository types
+var _ repository.Repository = &FileRepository{}
+var _ repository.WriteableRepository = &FileRepository{}
+var _ repository.HierarchicalRepository = &FileRepository{}
+var _ repository.ListableRepository = &FileRepository{}
+
+var _ fyne.URIReadCloser = &file{}
+var _ fyne.URIWriteCloser = &file{}
+
+type file struct {
+	*os.File
+	uri fyne.URI
+}
+
+func (f *file) URI() fyne.URI {
+	return f.uri
+}
+
+// FileRepository implements a simple wrapper around golang's filesystem
+// interface libraries. It should be registered by the driver on platforms
+// where it is appropriate to do so.
+//
+// This repository is suitable to handle file:// schemes.
+//
+// Since 2.0.0
+type FileRepository struct {
+}
+
+// NewFileRepository creates a new FileRepository instance. It must be
+// given the scheme it is registered for. The caller needs to call
+// repository.Register() on the result of this function.
+//
+// Since 2.0.0
+func NewFileRepository(scheme string) *FileRepository {
+	return &FileRepository{}
+}
+
+// Exists implements repository.Repository.Exists
+//
+// Since 2.0.0
+func (r *FileRepository) Exists(u fyne.URI) (bool, error) {
+	p := u.Path()
+
+	_, err := os.Stat(p)
+	ok := false
+
+	if err == nil {
+		ok = true
+	}
+
+	if os.IsNotExist(err) {
```

Comment:
nit: no need to check a nil error
```suggestion
	} else if os.IsNotExist(err) {
```

Reply:
26e7a0e31c5018fff46f5d15980ee52527291fd2